### ISSUE TITLE
bug(database): wal-e rewrote backup

### ISSUE THREAD

**Issue by aledbf** [deis/deis#3276]
The node where the database was running was rebooted and after ir started in another node, all the apps (like 50) were not there.
This is the log:
```
Mar 11 20:55:21 nodo-2 sh[6169]: 2015-03-11T20:55:21Z 2e9ec5be7ae0 confd[49]: ERROR exit status 127
Mar 11 20:55:21 nodo-2 sh[6169]: database: waiting for confd to write initial templates...
Mar 11 20:55:21 nodo-2 sh[6169]: 2015-03-11T20:55:21Z 2e9ec5be7ae0 confd[49]: ERROR exit status 127
Mar 11 20:55:26 nodo-2 sh[6169]: database: existing data directory found. Starting postgres...
Mar 11 20:55:26 nodo-2 sh[6169]: 2015-03-11 20:55:26 UTC LOG:  could not open usermap file "/var/lib/postgresql/main/pg_ident.conf": No such file or directory
Mar 11 20:55:26 nodo-2 sh[6169]: 2015-03-11 20:55:26 UTC LOG:  database system was interrupted; last known up at 2015-03-11 20:54:35 UTC
Mar 11 20:55:26 nodo-2 sh[6169]: 2015-03-11 20:55:26 UTC LOG:  database system was not properly shut down; automatic recovery in progress
Mar 11 20:55:26 nodo-2 sh[6169]: 2015-03-11 20:55:26 UTC LOG:  redo starts at 71/D7000090
Mar 11 20:55:26 nodo-2 sh[6169]: 2015-03-11 20:55:26 UTC LOG:  record with zero length at 71/D701B858
Mar 11 20:55:26 nodo-2 sh[6169]: 2015-03-11 20:55:26 UTC LOG:  redo done at 71/D701B828
Mar 11 20:55:26 nodo-2 sh[6169]: 2015-03-11 20:55:26 UTC LOG:  last completed transaction was at log time 2015-03-11 20:54:45.108194+00
Mar 11 20:55:26 nodo-2 sh[6169]: 2015-03-11 20:55:26 UTC LOG:  autovacuum launcher started
Mar 11 20:55:26 nodo-2 sh[6169]: 2015-03-11 20:55:26 UTC LOG:  database system is ready to accept connections
Mar 11 20:55:27 nodo-2 sh[6169]: ALTER ROLE
Mar 11 20:55:27 nodo-2 sh[6169]: ALTER ROLE
Mar 11 20:55:27 nodo-2 sh[6169]: database: postgres is running...
Mar 11 20:55:27 nodo-2 sh[6169]: 2015-03-11 20:55:27 UTC ERROR:  role "deis" already exists
Mar 11 20:55:27 nodo-2 sh[6169]: 2015-03-11 20:55:27 UTC STATEMENT:  CREATE ROLE deis WITH LOGIN;
Mar 11 20:55:27 nodo-2 sh[6169]: ERROR:  role "deis" already exists
Mar 11 20:55:27 nodo-2 sh[6169]: 2015-03-11 20:55:27 UTC ERROR:  database "deis" already exists
Mar 11 20:55:27 nodo-2 sh[6169]: 2015-03-11 20:55:27 UTC STATEMENT:  CREATE DATABASE deis WITH OWNER deis;
Mar 11 20:55:27 nodo-2 sh[6169]: ERROR:  database "deis" already exists
Mar 11 20:56:27 nodo-2 sh[6169]: wal_e.main   INFO     MSG: starting WAL-E
Mar 11 20:56:27 nodo-2 sh[6169]: DETAIL: The subcommand is "wal-push".
Mar 11 20:56:27 nodo-2 sh[6169]: STRUCTURED: time=2015-03-11T20:56:26.820415-00 pid=287
Mar 11 20:56:27 nodo-2 sh[6169]: wal_e.worker.upload INFO     MSG: begin archiving a file
Mar 11 20:56:27 nodo-2 sh[6169]: DETAIL: Uploading "pg_xlog/0000001600000071000000D7" to "s3://db_wal/wal_005/0000001600000071000000D7.lzo".
Mar 11 20:56:27 nodo-2 sh[6169]: STRUCTURED: time=2015-03-11T20:56:26.846134-00 pid=287 action=push-wal key=s3://db_wal/wal_005/0000001600000071000000D7.lzo prefix= seg=0000001600000071000000D7 state=begin
Mar 11 20:56:27 nodo-2 sh[6169]: wal_e.worker.upload INFO     MSG: completed archiving to a file
Mar 11 20:56:27 nodo-2 sh[6169]: DETAIL: Archiving to "s3://db_wal/wal_005/0000001600000071000000D7.lzo" complete at 7326.81KiB/s.
Mar 11 20:56:27 nodo-2 sh[6169]: STRUCTURED: time=2015-03-11T20:56:27.086045-00 pid=287 action=push-wal key=s3://db_wal/wal_005/0000001600000071000000D7.lzo prefix= rate=7326.81 seg=0000001600000071000000D7 state=complete
```

**Comment by bacongobbler** [comment]
> Mar 11 20:55:26 nodo-2 sh[6169]: 2015-03-11 20:55:26 UTC LOG:  database system was interrupted; last known up at 2015-03-11 20:54:35 UTC
> Mar 11 20:55:26 nodo-2 sh[6169]: 2015-03-11 20:55:26 UTC LOG:  database system was not properly shut down; automatic recovery in progress
> Mar 11 20:55:26 nodo-2 sh[6169]: 2015-03-11 20:55:26 UTC LOG:  redo starts at 71/D7000090
> Mar 11 20:55:26 nodo-2 sh[6169]: 2015-03-11 20:55:26 UTC LOG:  record with zero length at 71/D701B858
> Mar 11 20:55:26 nodo-2 sh[6169]: 2015-03-11 20:55:26 UTC LOG:  redo done at 71/D701B828
> Mar 11 20:55:26 nodo-2 sh[6169]: 2015-03-11 20:55:26 UTC LOG:  last completed transaction was at log time 2015-03-11 20:54:45.108194+00

Hmm, that "seems" to look like a regular restoration. Do you still have the WAL backups in Ceph available?

**Comment by aledbf** [comment]
@bacongobbler yes but the last record from api_app is from "2015-02-20 19:32:19"

**Comment by carmstrong** [comment]
What does the database log look like on the old host? Was it successfully writing WALs?

**Comment by carmstrong** [comment]
I'm starting to wonder if we should have database write to CephFS for its data, and just use WAL-e for backups -- it would never need to replay from them, since it could fail over and start back up immediately based on filesystem data.

**Comment by carmstrong** [comment]
I think this is addressed by #3128.
